src/main/java/uk/gov/hmcts/reform/iacaseapi/infrastructure/health/DownStreamHealthIndicator.java

```diff
@@ -0,0 +1,45 @@
+package uk.gov.hmcts.reform.iacaseapi.infrastructure.health;
+
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
+import org.springframework.boot.actuate.health.CompositeHealthContributor;
+import org.springframework.boot.actuate.health.HealthContributor;
+import org.springframework.boot.actuate.health.NamedContributor;
+import org.springframework.stereotype.Component;
+import org.springframework.web.client.RestTemplate;
+import uk.gov.hmcts.reform.iacaseapi.infrastructure.config.HealthCheckConfiguration;
+
+@Component
+public class DownStreamHealthIndicator implements CompositeHealthContributor {
+
+    private final HealthCheckConfiguration healthCheckConfiguration;
+
+    private Map<String, HealthContributor> contributors = new HashMap<>();
+
+    public DownStreamHealthIndicator(HealthCheckConfiguration healthCheckConfiguration) {
+        this.healthCheckConfiguration = healthCheckConfiguration;
+
+        healthCheckConfiguration.getServices().entrySet().stream()
+            .forEach(s -> {
+                contributors
+                    .put(s.getKey(), new ServiceHealthIndicator(
+                        s.getValue().get("uri"),
+                        s.getValue().get("response"),
+                        new RestTemplate()));
```

Comment:
Could you use the RestTemplate already configured in the app and inject instead?  Do we really need to call the `new` method here?

Reply:
Using restTemplate now, had to amend the RestTemplateConfiguration (change default message converter) in order to function correctly.